features/support/twopence_init.rb

```diff
@@ -1,50 +1,57 @@
 require 'twopence'
 
-# initialize ssh targets from environment variables.
-
-raise 'Server ip var empty' if ENV['TESTHOST'].nil?
-raise 'client ip var empty' if ENV['CLIENT'].nil?
-raise 'minion ip var empty' if ENV['MINION'].nil?
-raise 'ceos-minion ip var empty' if ENV['CENTOSMINION'].nil?
-raise 'ssh-minion ip var empty' if ENV['SSHMINION'].nil?
+# Initialize SSH targets from environment variables
+# If there is no proxy, use "None" for $PROXY environment variable
+raise 'Server IP address or domain name variable empty' if ENV['TESTHOST'].nil?
+raise 'Proxy IP address or domain name variable empty' if ENV['PROXY'].nil?
+raise 'Client IP address or domain name variable empty' if ENV['CLIENT'].nil?
+raise 'Minion IP address or domain name variable empty' if ENV['MINION'].nil?
+raise 'CentOS minion IP address or domain name variable empty' if ENV['CENTOSMINION'].nil?
+raise 'SSH minion IP address or domain name variable empty' if ENV['SSHMINION'].nil?
 
 $server_ip = ENV['TESTHOST']
+$proxy_ip = ENV['PROXY']
 $client_ip = ENV['CLIENT']
 $minion_ip = ENV['MINION']
 $ceos_minion_ip = ENV['CENTOSMINION']
 $ssh_minion_ip = ENV['SSHMINION']
 
-# define twopence object.
+# Define twopence objects
 $client = Twopence.init("ssh:#{$client_ip}")
+$proxy = Twopence.init("ssh:#{$proxy_ip}") if $proxy_ip != 'None'
 $server = Twopence.init("ssh:#{$server_ip}")
 $minion = Twopence.init("ssh:#{$minion_ip}")
 $ceos_minion = Twopence.init("ssh:#{$ceos_minion_ip}")
 $ssh_minion = Twopence.init("ssh:#{$ssh_minion_ip}")
 
-# lavanda library module extension.
-# look at support/lavanda.rb for more details
+# Lavanda library module extension
+# Look at support/lavanda.rb for more details
 $server.extend(LavandaBasic)
+$proxy.extend(LavandaBasic) if $proxy
 $client.extend(LavandaBasic)
 $minion.extend(LavandaBasic)
 $ceos_minion.extend(LavandaBasic)
 $ssh_minion.extend(LavandaBasic)
 
-# add here new vms .
+# Initialize hostname
 nodes = [$server, $client, $minion, $ceos_minion, $ssh_minion]
+nodes.push($proxy) if $proxy
```

Comment:
i think something like this is more consistent, and more generic.
```console
nodes = [server, client, minion, ceos_minion, ssh_minion, proxy]
nodes.each do |node|
      # if node false just skip
      next unless node
end
```

Reply:
ok

Reply:
Done.